swagger-api/swagger-ui
# feat: add modelcollapse to primitive models

Add collapse to primitive models
### Description

Now `primitive-model` is wrapped inside a `ModelCollapse` component to let the user expand and collapse all the content of each primitive models such as (string, number, integer and boolean)

### Motivation and Context





This is a fix to the following issue number https://github.com/swagger-api/swagger-ui/issues/7549

### How Has This Been Tested?




- Create a new test.yaml file inside ./dev-helpers/ and paste the yaml code that is in the issue link. (https://github.com/swagger-api/swagger-ui/issues/7549)
- In ./dev-helpers/index.html change url parameter of ui variable to: ./dev-helpers/test.yaml
- run npm run dev in terminal
- Open http://localhost:3200/
- Check if TypeEnum can expand and collapse



### Screenshots (if appropriate):


https://user-images.githubusercontent.com/12849040/137395252-bee69704-d144-46c8-b28c-6967d90fe56c.mov



## Checklist



### My PR contains... 


### My changes...

### Documentation

### Automated tests
